Navigation Bar

khaledsaadeh opened the pull request:
Added the navigation bar with search functionality, search has a minor bug that will be handled in a later PR

hazemturki commented on src/components/NavBar.js:
```diff
@@ -0,0 +1,47 @@
+import { useState, useEffect } from "react";
+import axios from "axios";
+const NavBar = (props) => {
+  const API =
+    "https://api.themoviedb.org/3/search/movie?api_key=4f298a53e552283bee957836a529baec&language=en-US&page=1&include_adult=false&query=";
```
Not sure this is the best place for such a constant.
Doesn't seem appropriate, this is a NavBar component it shouldn't define a API URL

hazemturki commented on src/components/NavBar.js:
```diff
@@ -0,0 +1,47 @@
+  const searchHandler = (event) => {
+    event.preventDefault();
+    if (search.trim() !== "") {
+      axios.get(API + search).then((res) => {
```
Please don't use simple string concatenation to add the query and instead use Axios.get options.
Please refer to the official axios docs

```suggestion
      axios.get(API, {params: search}).then((res) => {
```

hazemturki commented on src/components/NavBar.js:
```diff
@@ -0,0 +1,47 @@
+  const [search, setSearch] = useState("");
+  
+  const [movies, setMovies] = useState([]);  
+  const searchHandler = (event) => {
```
This search handler should trigger a callBackFunction to the app, but it shouldn't be responsible for fetching data.

hazemturki commented on src/App.js:
```diff
@@ -14,10 +15,16 @@ function App() {
   return (
     <div className="App">
-      <MovieList movies={movies}  />
+      <NavBar passToApp={showSearchResults} />
```
Pass to app is an ambiguous name for a prop.
Please rename this according to the other comments in the review.

```suggestion
      <NavBar ?={showSearchResults} />
```

khaledsaadeh commented:
-Added API constants to the main component.
-Used the {params} option from axios.get instead of string concatenation.
-Search handler now triggers a function responsible for fetching the data, instead of fetching them itself.
-changed the passToApp prop name to a more appropiate prop name.
